### PR TITLE
Display HTML block translation field as a text area in CTE

### PR DESCRIPTION
**Pull request by mrudzki91**
https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-7310

### CODE REVIEW

**Review by andreasciamanna — DISMISSED** [review]

File: src/strings-in-block/dom-handler/html-block.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace WPML\PB\Gutenberg\StringsInBlock\DOMHandler;
+
+class HtmlBlock extends DOMHandle {
```
Comment (strategio) [inline src/strings-in-block/dom-handler/html-block.php]: @mrudzki91, if you extend `HtmlBlock` from `StandardBlock`, you just have to write the method `getInnerHTML`.

File: src/strings-in-block/class-html.php [hunk 2]
```diff
@@ -162,12 +164,15 @@ private function get_block_queries( \WP_Block_Parser_Block $block ) {
...
-	 * @return ListBlock|StandardBlock
+	 * @return ListBlock|StandardBlock|HtmlBlock
 	 */
 	private function get_dom_handler( \WP_Block_Parser_Block $block ) {
 		if ( self::LIST_BLOCK_NAME === $block->blockName ) {
 			return new ListBlock();
 		}
+		if ( self::HTML_BLOCK_NAME === $block->blockName ) {
+			return new HtmlBlock();
+		}
```
Comment (strategio) [inline src/strings-in-block/class-html.php]: @mrudzki91, I suggest this smart pattern to replace kind of switch:

```php
$class = wpml_collect( [
    self::LIST_BLOCK_NAME => ListBlock::class,
    self::HTML_BLOCK_NAME => HtmlBlock::class,
] )->get( $block->blockName, StandardBlock::class );

return new $class();
```

Credits: @brucepearson :)

**Review by strategio — COMMENTED** [review]
@mrudzki91, I have 2 small suggestions to reduce the quantity of code and duplication.

File: src/strings-in-block/dom-handler/html-block.php (hunk 1 above)
Comment (mrudzki91) [reply src/strings-in-block/dom-handler/html-block.php]: @strategio you right, it's done

File: src/strings-in-block/class-html.php (hunk 2 above)
Comment (mrudzki91) [reply src/strings-in-block/class-html.php]: @strategio great idea, thanks! :)

**Review by strategio — APPROVED** [review]
+1 nice job @mrudzki91

**Review by brucepearson — APPROVED** [review]

**Pull request merged**